types/rdf-ext/lib/Dataset.d.ts

```diff
@@ -37,6 +37,9 @@ export interface DatasetExt extends Dataset<QuadExt> {
     toStream(): Stream<QuadExt>;
 
     equals(other: DatasetCore): boolean;
+
+    reduce(callbackfn: (previousValue: QuadExt, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => QuadExt, initialValue?: QuadExt): QuadExt;
+    reduce<U>(callbackfn: (previousValue: U, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => U, initialValue: U): U;
```

Comment:
Why not

```suggestion
    reduce<U = QuadExt>(callbackfn: (previousValue: U, currentValue: QuadExt, currentIndex: number, self: DatasetExt) => U, initialValue: U): U;
```

Reply:
The overload with type argument requires an initial value. Only in the non-generic version it can be skipped in which case the first element will be used as `previousValue` in first iteration